### PR TITLE
Feature/add settings

### PR DESCRIPTION
**Pull request by lcgao**
- [Add a settings screen](https://roboteam.atlassian.net/browse/ROBO-10177)
- [Change tips](https://roboteam.atlassian.net/browse/ROBO-10173)
- [Remove movement badge](https://roboteam.atlassian.net/browse/ROBO-10176)

### CODE REVIEW

File: app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    settings.isUsingVoiceGreeter = false
+                }
+                saveSettings(settings)
+            }
+        })
+        customToggleVoiceGreeter.setToggleListener(object : CustomToggle.ToggleListener {
+            override fun onToggle(on: Boolean) {
+                val settings = getSettings()
+                settings.isUsingVoiceGreeter = on
+                saveSettings(settings)
```
Comment (hermanShai) [inline app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt]: minimise to one line
saveSettings(getSettings().apply{isUsingVoiceGreeter = on})

File: app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            }
+        })
+        relativeLayoutDefaultMessage.setOnClickListener {
+            val settings = getSettings()
+            settings.isUsingDefaultMessage = true
+            saveSettings(settings)
```
Comment (hermanShai) [inline app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt]: same

File: app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        }
+        relativeLayoutCustom.setOnClickListener {
+            val settings = getSettings()
+            settings.isUsingDefaultMessage = false
+            activityCallback.startFragment(CustomGreeterFragment.newInstance())
+            saveSettings(settings)
```
Comment (hermanShai) [inline app/src/main/java/com/robotemi/welcomingbtob/settings/ConfigurationFragment.kt]: same

File: app/src/main/java/com/robotemi/welcomingbtob/settings/CustomGreeterFragment.kt [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? = inflater.inflate(R.layout.fragment_custom, container, false)
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val customizeGreeter = getSettings().customMessage
+        activityCallback.setTitle(getString(R.string.fragment_custom_greeter))
```
Comment (hermanShai) [inline app/src/main/java/com/robotemi/welcomingbtob/settings/CustomGreeterFragment.kt]: use apply

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -35,6 +33,24 @@
...
+        </activity>
+
+        <activity
+            android:name=".settings.SettingsActivity"
+            android:screenOrientation="landscape">
+            <meta-data
+                android:name="com.robotemi.sdk.metadata.UI_MODE"
+                android:value="1" />
+            <meta-data
+                android:name="com.robotemi.sdk.metadata.UI_FLAG"
```
Comment (hermanShai) [inline app/src/main/AndroidManifest.xml]: why is this meta data duplicated with line 41?

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -35,6 +33,24 @@
...
+                android:value="1" />
+        </activity>
+
+        <activity
+            android:name=".settings.SettingsActivity"
+            android:screenOrientation="landscape">
+            <meta-data
+                android:name="com.robotemi.sdk.metadata.UI_MODE"
```
Comment (hermanShai) [inline app/src/main/AndroidManifest.xml]: why is this meta data duplicated with line 38?

File: app/src/main/java/com/robotemi/welcomingbtob/settings/CustomGreeterFragment.kt [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        })
+    }
+
+    private fun saveCustomizeGreeterMessage(text: String) {
+        val settings = getSettings()
+        settings.customMessage = text
+        saveSettings(settings)
+    }
+
+    private fun getSettings(): SettingsModel {
```
Comment (hermanShai) [inline app/src/main/java/com/robotemi/welcomingbtob/settings/CustomGreeterFragment.kt]: use getSettings() = SettingsModel.getSettings(context!!)

File: app/src/main/AndroidManifest.xml (hunk 5 above)
Comment (lcgao) [reply app/src/main/AndroidManifest.xml]: These meta-data is for Activity only.

File: app/src/main/AndroidManifest.xml (hunk 6 above)
Comment (lcgao) [reply app/src/main/AndroidManifest.xml]: These meta-data is for Activity only.

**Pull request merged**